extensions/ray_objectives.py

```diff
@@ -0,0 +1,41 @@
+import logging
+import ray
+from collections import deque
+
+ACTOR_NAME="BabyAGI Objectives"
+
+try:
+    ray.init(address="auto", namespace="babyagi", logging_level=logging.FATAL, ignore_reinit_error=True)
```

Comment:
Not familiar with ray, but what's the point of a try and except double inits?

Reply:
From the docs:
2. If no address is provided, try to find an existing Ray instance to connect to. This is done by first checking the environment variable RAY_ADDRESS. If this is not defined, check the address of the latest cluster started (found in /tmp/ray/ray_current_cluster) if available. If this is also empty, then start a new local Ray instance. 3. If the provided address is “auto”, then follow the same process as above. However, if there is no existing cluster found, this will throw a ConnectionError instead of starting a new local Ray instance.

So it first tries to connect to an existing cluster automatically. If that fails, it excepts and tries the same again, but this time starting a new cluster if no existing one is found.

This feels redundant to me; I guess the try-except block could be removed and the second init kept.

Reply:
interesting, would be helpful for a comment as it's hard to tell what's happening here.

Reply:
For some reason second instance wouldn't connect to the cluster started by the first, that's why I had to do it this way. Either for some reason I have older Ray library, or the docs are... lying...